File: app/src/main/java/com/android/mathias/velocity/Walk.java
package com.android.mathias.velocity;

import java.util.Date;

class Walk {
    private long mId;
    private Route mRoute;
    private long mDuration;
    private Date mDate;

    public Walk () {
        mRoute = new Route("NOT SET");
        mDuration = 0;
        mDate = new Date();
    }

    Walk (long duration, Date date, Route route) {
        mRoute = route;
        mDuration = duration;
        mDate = date;
    }

    Walk (long id, long duration, Date date, Route route) {
        mId = id;
        mRoute = route;
        mDuration = duration;
        mDate = date;
    }

    long getId() {
        return mId;
    }
    void setId(long id) {
        mId = id;
    }

    Route getRoute() {
        return mRoute;
    }
    void setRoute(Route route) {
        mRoute = route;
    }

    long getDuration() {
        return mDuration;
    }
    void setDuration(long duration) {
        mDuration = duration;
    }

    Date getDate() {
        return mDate;
    }
    void setDate(Date date) {
        mDate = date;
    }
}

File: app/src/main/java/com/android/mathias/velocity/Walk.kt
package com.android.mathias.velocity

import java.util.*

internal class Walk {
    var id: Long = 0
    var route: Route? = null
    var duration: Long = 0
    var date: Date? = null

    constructor() {
        route = Route("NOT SET")
        duration = 0
        date = Date()
    }

    constructor(duration: Long, date: Date, route: Route) {
        this.route = route
        this.duration = duration
        this.date = date
    }

    constructor(id: Long, duration: Long, date: Date, route: Route) {
        this.id = id
        this.route = route
        this.duration = duration
        this.date = date
    }
}
